lib/update/update_supporter.rb

```diff
@@ -18,7 +18,8 @@ def self.bulk_delete(np_id, supporter_ids)
   end
 
   def self.general_info(supporter_id, data)
-    Qx.update(:supporters).set(data).where(id: supporter_id).returning('*').ex.last
+    supporter = Supporter.find(supporter_id)
+    supporter.update_attributes(data)
```

Comment:
Usually better to use `#update` instead of `#update_attributes`. They do the same thing but `#update_attributes` is deprecated in later Rails versions.